src/main/java/de/dennisguse/opentracks/services/TrackDeleteServiceConnection.java

```diff
@@ -0,0 +1,68 @@
+package de.dennisguse.opentracks.services;
+
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.ServiceConnection;
+import android.os.IBinder;
+
+import androidx.annotation.NonNull;
+
+import java.util.ArrayList;
+
+import de.dennisguse.opentracks.BuildConfig;
+import de.dennisguse.opentracks.content.data.Track;
+
+import static de.dennisguse.opentracks.services.TrackDeleteService.EXTRA_TRACK_IDS;
+
+public class TrackDeleteServiceConnection implements ServiceConnection {
```

Comment:
Same as TrackRecordingServiceConnection, right?
Why not use ServiceConnection<T extends Service> as a base class?

Reply:
> Same as TrackRecordingServiceConnection, right?

Yes, the approach is the same, but the implementation of both connections are different.